src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -595,8 +595,14 @@ public function formAction($id, Request $request)
             // this controller can be called as an AJAX JSON route or a HTML page
             // so we need to return the right type of response if an exception it thrown
             if ($request->isXmlHttpRequest()) {
+                $error = [];
+
+                if ($this->get('prestashop.adapter.environment')->isDebug()) {
+                    $error['error'] = $e->getMessage();
+                }
+
                 return $this->returnErrorJsonResponse(
-                    [],
+                    $error,
```

Comment:
Hi @PrestaEdit. What do you think about making a custom message instead of returning the exception message (which may return some critical informations) ?

Reply:
I think that the goal of this pull request is to get the real error message.

You already get a notice about the update failed of the form, but you need (as you are in Back Office AND in debug mode) to get the real exception to allow you to know where the error is located.